File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) return false;

        int[] occurences = new int[26];

        for(int i = 0; i < occurences.length; i++) {
            occurences[i] = 0;
        }

        for(int i = 0; i < s.length(); i++) {
            occurences[s.charAt(i) - 'a'] += 1;
        }

        for(int i = 0; i < t.length(); i++) {
            occurences[t.charAt(i) - 'a'] -= 1;
        }

        int sumArray = 0;
        for(int i = 0; i < occurences.length; i++) {
            if (occurences[i] != 0) return false;
        }

        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        int arr[26];

        for (char letter : s) {
            arr[static_cast<int>(letter) - static_cast<int>('a')]++;
        }

        for (char letter : t) {
            arr[static_cast<int>(letter) - static_cast<int>('a')]--;
        }

        for (int i = 0; i < 26; i++) {
            if(arr[i] != 0) return false;
        }

        return true;
    }
};